core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -522,6 +545,12 @@ public SELF withExtraHost(String hostname, String ipAddress) {
         return self();
     }
 
+    @Override
+    public SELF withNetworkMode(String networkMode) {
```

Comment:
Accept network mode as a string - this is not abstracted from the raw API value, and it is assumed that the user knows what they are doing.